rust-src/concordium_base/src/web3id/v1/proofs.rs

```diff
@@ -0,0 +1,1132 @@
+    fn verify(
+        &self,
+        global_context: &GlobalContext<C>,
+        transcript: &mut RandomOracle,
+        input: &CredentialVerificationMaterial<P, C>,
+    ) -> bool {
+        let CredentialVerificationMaterial::Identity(IdentityCredentialVerificationMaterial {
+            ip_info,
+            ars_infos,
+        }) = input
+        else {
+            // mismatch in types
+            return false;
+        };
+
+        let Ok(cred_id_data) = self.subject.cred_id.try_to_data() else {
+            return false;
+        };
+
+        let id_attr_cred_info = IdentityAttributesCredentialsInfo {
+            values: IdentityAttributesCredentialsValues {
+                ip_identity: self.issuer,
+                threshold: cred_id_data.threshold,
+                ar_data: cred_id_data.ar_data,
+                attributes: self.proof.proof.identity_attributes.clone(),
+                validity: self.validity.clone(),
+            },
+            proofs: self.proof.proof.identity_attributes_proofs.clone(),
+        };
+
+        if identity_attributes_credentials::verify_identity_attributes(
+            global_context,
+            IpContextOnly {
+                ip_info,
+                ars_infos: &ars_infos.anonymity_revokers,
+            },
+            &id_attr_cred_info,
+            transcript,
+        )
+        .is_err()
+        {
+            return false;
+        }
+
+        let cmm_attributes: BTreeMap<_, _> = self
+            .proof
+            .proof
+            .identity_attributes
+            .iter()
+            .filter_map(|(tag, attr)| match attr {
+                IdentityAttribute::Committed(cmm) => Some((*tag, *cmm)),
+                _ => None,
+            })
+            .collect();
+
+        verify_statements(
+            &self.subject.statements,
+            &self.proof.proof.statement_proofs,
+            &cmm_attributes,
+            global_context,
+            transcript,
+        )
+    }
+}
+
+fn verify_statements<
+    'a,
+    C: Curve,
+    AttributeType: Attribute<C::Scalar> + 'a,
+    TagType: Ord + crate::common::Serialize + 'a,
+>(
+    statements: impl IntoIterator<Item = &'a AtomicStatement<C, TagType, AttributeType>>,
+    proofs: impl IntoIterator<Item = &'a AtomicProof<C, AttributeType>>,
+    cmm_attributes: &BTreeMap<TagType, Commitment<C>>,
+    global_context: &GlobalContext<C>,
+    transcript: &mut RandomOracle,
+) -> bool {
+    statements.into_iter().zip_longest(proofs).all(|elm| {
+        elm.both().map_or(false, |(statement, proof)| {
+            statement.verify(
+                ProofVersion::Version2,
+                global_context,
+                transcript,
+                cmm_attributes,
+                proof,
+            )
+        })
+    })
+}
+
+impl<C: Curve, AttributeType: Attribute<C::Scalar>> AccountBasedSubjectClaims<C, AttributeType> {
+    /// Proof described in "15.4.1 Account Based Presentations" (blue paper v2.2.0)
```

Comment:
As people won't have (version v2.2.0) at hand in the future, I suggest, to link to a specific blue paper version by using the most recent git commit. 
https://github.com/Concordium/concordium.github.io/blob/8bccb0608f37a55ad1312dc32faf621b7b5b14ea/public/governance/bluepaper/concordium-bluepaper.pdf